src/utils.h

```diff
@@ -43,9 +43,11 @@
 #ifdef _WIN32
 # include <direct.h>
 # include <io.h>
+# include <BaseTsd.h>
 # ifndef S_ISDIR
 #  define S_ISDIR(m) (((m) & S_IFMT) == S_IFDIR)
 # endif
+typedef SSIZE_T ssize_t;
```

Comment:
does this one need to be squashed in somewhere or is it a standalone one? If the latter could you factor it out into a separate PR and merge it directly please? If the former, please squash it in :)